google/resumable_media/_upload.py

```diff
@@ -817,7 +817,7 @@ def get_next_chunk(stream, chunk_size, total_bytes):
             raise ValueError(msg)
 
     content_range = get_content_range(start_byte, end_byte, total_bytes)
-    return start_byte, end_byte, payload, content_range
+    return start_byte, max([end_byte, 0]), payload, content_range
```

Comment:
Also just confirmed this WON'T happen with `Blob.upload_from_file(..., size=None)` because it sets `stream_final=False`, so `ResumableUpload._total_bytes is None` (whereas `Blob.upload_from_filename` explicitly passes `size=os.fstat(file_obj.fileno()).st_size` to `upload_from_file`)